PetGame/Services/PetService.cs

```diff
@@ -200,5 +211,196 @@ public bool DeletePet(ulong petId, ulong ownerId)
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= RaceDecrease) > 0))
+                        {
+                            hungerPercentage -= RaceDecrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                    //else, decrease hunger
+                    else
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= HungerDecreasePerHour) > 0))
+                        {
+                            hungerPercentage -= HungerDecreasePerHour;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                }
+            }//end
+
+            //check the number of hours since the last activity
+            int HoursSinceLastActivity = 0;
+            int MinutesTillNextAction = 5;
+
+            //check to see if there were any activities in the specified interval
+            if (PastActivities.Count == 0)
+            {
+                HoursSinceLastActivity = HoursToCheck;
+            }
+            else
+            {
+                //get the last activity
+                Activity LastActivity = PastActivities[PastActivities.Count - 1];
+
+                //check the number of hours since the last activity
+                HoursSinceLastActivity = DateTime.Now.Hour - LastActivity.Timestamp.Hour;
+
+                //check the number of minute since the last activity
+                int MinutesSinceLastActivity = 0;
+                MinutesSinceLastActivity = DateTime.Now.Minute - LastActivity.Timestamp.Minute;
+
+                //if it's been more than 5 minutes, the user can perform another action
+                //otherwise, leave the default 5 min value in place
+                if (MinutesSinceLastActivity > 5)
+                {
+                    MinutesTillNextAction = 0;
+                }
+            }
+
+            //multiply HappinessDecrease by the number of hours to get the
+            //happiness percentage
+            happinessPercentage = (HoursSinceLastActivity * (-1 * HappinessDecreasePerHour));
+
+            //ensure that happiness cannot be <0 or >1
+            if (happinessPercentage < 0.0)
+            {
+                happinessPercentage = 0.0;
+            }
+            else if (happinessPercentage > 1.0)
+            {
+                happinessPercentage = 1.0;
+            }
+
+            //ensure that hunger cannot be <0 or >1
+            if (hungerPercentage < 0.0)
+            {
+                hungerPercentage = 0.0;
+            }
+            else if (hungerPercentage > 1.0)
+            {
+                hungerPercentage = 1.0;
+            }
```

Comment:
Again, you repeat yourself a lot here. I think you should implement a `double constrain(double value, double min, double max)` function, since you perform the same logic a few times here.